core/parentchain/light-client/src/io.rs

```diff
@@ -81,9 +87,11 @@ impl<B, L> LightClientStateSeal<B, L> {
 	}
 }
 
-impl<B: Block, LightClientState: Decode + Encode + Debug> LightClientSealing<LightClientState>
+impl<B: Block, LightClientState: Decode + Encode + Debug> LightClientSealing
 	for LightClientStateSeal<B, LightClientState>
 {
+	type LightClientState = LightClientState;
```

Comment:
Associated type is more flexible, as otherwise I would have to declare an extra generic param for the block, because the generic type of the state is generic over the block, which in turn means that I would have had to introduce a phantom type for the block, not funny.